src/include/OpenImageIO/imagebufalgo.h

```diff
@@ -649,10 +649,23 @@ bool OIIO_API resize (ImageBuf &dst, const ImageBuf &src, Filter2D *filter,
 /// needed, rather than attempting to interpolate deep pixels (regardless of
 /// the value of `interpolate`).
 ImageBuf OIIO_API resample (const ImageBuf &src, bool interpolate = true,
-                        ROI roi={}, int nthreads=0);
+                            ROI roi={},
+                            ImageBuf::WrapMode wrap=ImageBuf::WrapDefault,
+                            int nthreads=0);
 /// Write to an exsisting image `dst` (allocating if it is uninitialized).
 bool OIIO_API resample (ImageBuf &dst, const ImageBuf &src,
-                        bool interpolate = true, ROI roi={}, int nthreads=0);
+                        bool interpolate = true, ROI roi={},
+                        ImageBuf::WrapMode wrap=ImageBuf::WrapDefault,
+                        int nthreads=0);
+inline ImageBuf OIIO_API resample (const ImageBuf &src, bool interpolate,
```

Comment:
What you've done here adding inline functions with the old signatures is good, I'm glad you thought of that.

However, this only preserves source compatibility (with a recompile). That is totally fine if you intend for these changes to only be in master (i.e., "someday 2.2").

If you would like the new kind of resample/resize/fit to be backported to release branch 2.1, then we need to preserve link compatibility as well. For that, you'd need to make this here in the header just be the declaration, and then put the body (which still just trivially calls the new, more complete call) in the same source module as the existing one.

I hope you understand what I'm getting at. But again, if you don't need this backported, then this inline approach is fine.

Sorry, one more thing... Can you please move the "back compatibility" versions (whether they be inline or just declarations) down to the bottom of imagebufalgo.h? You can see there is already a big section with this comment: 

    // DEPRECATED(1.9): These are all ...

I would add these ones below it, with a brief comment:

    // DEPRECATED(2.2): moving forward, please favor the ones with a 'wrap' parameter.